### PR TITLE
dev/focus-cleanup

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Change Request

## Overview

Cleaned up focus result a bit. We were inappropriately checking if it was null (should never be null)

## Target of the change:

Is this enhancement for:

- Core (core framework, interfaces and definitions)

## Changes:

- Removed a flag to set the pointer's `IsTargetPositionLockedOnFocusLock` to true if our `hitObject` is not the current pointer target.
- a bit of formatting
- Removed null checks for pointer.Result. This value should never be null.

### CODE REVIEW

**Comment by StephenHodgson** [comment]
@SimonDarksideJ not as confident in these changes, so I'd like a good review and test before we merge these

File: XRTK-Core/Packages/com.xrtk.core/Services/InputSystem/FocusProvider.cs [hunk 1]
```diff
@@ -190,14 +190,6 @@ public PointerData(IMixedRealityPointer pointer)
 
             public void UpdateHit(PointerHitResult hitResult)
             {
-                if (hitResult.HitObject != CurrentPointerTarget)
-                {
-                    // Pointer.OnPreCurrentPointerTargetChange();
-
-                    // Set to default:
-                    Pointer.IsTargetPositionLockedOnFocusLock = true;
```
Comment (SimonDarksideJ) [inline XRTK-Core/Packages/com.xrtk.core/Services/InputSystem/FocusProvider.cs]: So what sets this to true now, locking it to a specific target?

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Services/InputSystem/FocusProvider.cs]: In the inspectors or the developer toggles this flag in one of their scrips

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Services/InputSystem/FocusProvider.cs]: This didn't look like it was needed. I'm not entirely sure what it was there to begin with

Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Services/InputSystem/FocusProvider.cs]: If we are unsure, then we should poll the community and even query it on Mrtk as to its use case and validity. As we do with any original code

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Services/InputSystem/FocusProvider.cs]: I think you missed the point.

This doesn't actually do anything afaik. I'm not interested in how the MRTK does it, or its usage there as it's likely been changed.

**Comment by StephenHodgson** [comment]
Gonna revisit this

**Pull request closed without merging**